src/time.rs

```diff
@@ -0,0 +1,258 @@
+use crate::sys::time::TimeSpec;
+#[cfg(any(
+    target_os = "freebsd",
+    target_os = "dragonfly",
+    target_os = "linux",
+    target_os = "android",
+    target_os = "emscripten",
+))]
+use crate::{unistd::Pid, Error};
+use crate::{Errno, Result};
+use libc::{self, clockid_t};
+use std::mem::MaybeUninit;
+
+/// Clock identifier
+///
+/// Newtype pattern around `clockid_t` (which is just alias). It pervents bugs caused by
+/// accidentally passing wrong value.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct ClockId(clockid_t);
```

Comment:
Should this use `libc_enum!`?

Reply:
The enum won't work for `clockid_t` returned by `clock_getcpuclockid`, that's why I had to go with struct.

Reply:
You're worried that the OS may add a new clockid that Nix doesn't know about?  Ok, that's a fair reason.